woocommerce/woocommerce
# Move template addition to feature flag

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

This PR removes the `template` and `templateLock` configs added in 7.6.1 by default and moves them to a filter, only added when a feature flag is enabled, as the template is needed for the new product block editor.

Closes #38269



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Enable Gutenberg within the product block editor by either enabling this plugin: https://wordpress.org/plugins/blocks-product-editor-for-woocommerce/ or using this filter
```
function wplook_activate_gutenberg_products($can_edit, $post_type)
{
    if ($post_type == 'product') {
        $can_edit = true;
    }

    return $can_edit;
}
add_filter('use_block_editor_for_post_type', 'wplook_activate_gutenberg_products', 100, 2);
```
2. Go to **Products > Add New** and notice how the block editor loads correctly for the product description
3. Edit the product title and add some new blocks and saving the product (this should work correctly).
4. Enable the `product-block-editor` feature flag using the WooCommerce Beta Tester plugin
5. Refresh the page and go to **Products > Add New**
6. The new product block editor should render correctly and allow you to add and save products.
